### PR TITLE
 Add span_id to discover and promote it in discover, errors dataset

### PR DESCRIPTION
**Pull request by vladkluev**
## Problem
Discover queries by span_id where unwrapping the context column meaning we needed to store it.

DDL: 
```
ALTER TABLE discover_local ADD COLUMN IF NOT EXISTS span_id Nullable(UInt64) AFTER contexts;
```

### CODE REVIEW

File: snuba/clickhouse/translators/snuba/mappers.py [hunk 1]
```diff
@@ -24,6 +24,7 @@
     String,
 )
 
+
```
Comment (evanh) [inline snuba/clickhouse/translators/snuba/mappers.py:27]: 💀

**Review by evanh — COMMENTED** [review]
Please look at `BaseTypeConverter` and the classes that extend it if you haven't already. I think this is what you want.

File: snuba/datasets/storages/discover.py [hunk 2]
```diff
@@ -82,7 +83,7 @@
                     "sentry:dist": "dist",
                     "sentry:user": "user",
                 },
-                "contexts": {"trace.trace_id": "trace_id"},
+                "contexts": {"trace.trace_id": "trace_id", "trace.span_id": "span_id"},
```
Comment (fpacifici) [inline snuba/datasets/storages/discover.py:86]: Please separate the migration part of the PR from any code that would start using the column. 
There is no way to synchronize migration and application change in production. So first we need to apply the migration and then we need to start using the column, or we need to hold on on the PR (which then can be modified) while we apply the production migration.

**Review by fpacifici — CHANGES_REQUESTED** [review]

File: snuba/clickhouse/translators/snuba/mappers.py (hunk 1 above)
Comment (vladkluev) [reply snuba/clickhouse/translators/snuba/mappers.py:27]: I had touched that file (but removed the changes) but black autoformatted it anyways.  If I delete this line, black will put it right back

File: snuba/datasets/storages/discover.py (hunk 2 above)
Comment (vladkluev) [reply snuba/datasets/storages/discover.py]: good spot thank you

File: tests/datasets/test_discover.py [hunk 3]
```diff
@@ -121,8 +122,8 @@
     ),
     # # No conditions, other referenced columns
     ({"selected_columns": ["group_id"]}, EntityKey.DISCOVER_EVENTS),
-    ({"selected_columns": ["span_id"]}, EntityKey.DISCOVER_TRANSACTIONS),
-    ({"selected_columns": ["group_id", "span_id"]}, EntityKey.DISCOVER),
+    ({"selected_columns": ["span_id"]}, EntityKey.DISCOVER),
```
Comment (vladkluev) [inline tests/datasets/test_discover.py:125]: `span_id` is now in the common columns and thus we would hit the merge table

File: tests/datasets/test_discover.py [hunk 4]
```diff
@@ -121,8 +122,8 @@
...
+    ({"selected_columns": ["group_id", "span_id"]}, EntityKey.DISCOVER_EVENTS),
```
Comment (vladkluev) [inline tests/datasets/test_discover.py:126]: `group_id` is in `DISCOVER_EVENTS`, `span_id` is in the common columns, thus we can hit `DISCOVER_EVENTS`

**Review by lynnagara — APPROVED** [review]

**Pull request merged**